### PR TITLE
Switch to path-based i18n routing

### LINKED ISSUE #700: Translations aren't shown on localized homepages
## Description

If you visit a localized site, the homepage isn't translated even though translations exist.

## Reproduction

1. Visit https://es.wordpress.org/openverse
2. Observe the page is in English

## Screenshots

CleanShot 2022-01-24 at 22 26 42@2x

## Additional context

The current cookie-based translation mechanism is also the only response cookie sent by Openverse. We could eliminate this cookie based approach and switch to a path approach, which would _also_ enable more aggressive caching on the frontend, as we could cache each localized page without worrying about, for example, translations in one language polluting the cache. 

## Resolution

- [x] 🙋 I would be interested in resolving this bug.

### PR DESCRIPTION
**Pull request by zackkrida**
## Fixes

Fixes #700 by @zackkrida 

## Description

This PR switches Openverse to use the url path instead of a cookie for setting translations. This has a few benefits:

1. Don't have to worry about cookie values being set correctly when switching from one locale to another
2. Makes it easy for developers to test different locales, just modify the url path
3. Allows for much more aggressive front-end caching (currently the site is 'dynamic' in the sense that the same page with the same url should show different contents based on a cookie). This makes each page infinitely cacheable by url.
4. Fixes the issue in #700 

To use this PR, we need an accompanying PR in the Openverse WordPress theme that prefixes the iframe url with the correct locale string. 

## Testing Instructions

1. Run the app locally
2. Try different locale-based paths like `localhost:8443/es`, `localhost:8443/ru` and so on.

## Checklist

- [x] My pull request has a descriptive title (not a vague title like `Update index.md`).
- [x] My pull request targets the *default* branch of the repository (`main`) or a parent feature branch.
- [x] My commit messages follow [best practices][best_practices].
- [x] My code follows the established code style of the repository.
- [ ] I added or updated tests for the changes I made (if applicable).
- [x] I added or updated documentation (if applicable).
- [x] I tried running the project locally and verified that there are no visible errors.

[best_practices]:https://git-scm.com/book/en/v2/Distributed-Git-Contributing-to-a-Project#_commit_guidelines

## Developer Certificate of Origin

<details>
<summary>Developer Certificate of Origin</summary>

```
Developer Certificate of Origin
Version 1.1

Copyright (C) 2004, 2006 The Linux Foundation and its contributors.
1 Letterman Drive
Suite D4700
San Francisco, CA, 94129

Everyone is permitted to copy and distribute verbatim copies of this
license document, but changing it is not allowed.


Developer's Certificate of Origin 1.1

By making a contribution to this project, I certify that:

(a) The contribution was created in whole or in part by me and I
    have the right to submit it under the open source license
    indicated in the file; or

(b) The contribution is based upon previous work that, to the best
    of my knowledge, is covered under an appropriate open source
    license and I have the right under that license to submit that
    work with modifications, whether created in whole or in part
    by me, under the same open source license (unless I am
    permitted to submit under a different license), as indicated
    in the file; or

(c) The contribution was provided directly to me by some other
    person who certified (a), (b) or (c) and I have not modified
    it.

(d) I understand and agree that this project and the contribution
    are public and that a record of the contribution (including all
    personal information I submit with it, including my sign-off) is
    maintained indefinitely and may be redistributed consistent with
    this project or the open source license(s) involved.
```

</details>

### CODE REVIEW

File: nuxt.config.js [hunk 1]
```diff
@@ -172,20 +172,12 @@ export default {
...
      * */
-    detectBrowserLanguage: {
-      useCookie: true,
-      cookieKey: 'i18n_redirected',
-      alwaysRedirect: true,
-      cookieCrossOrigin: true,
-      cookieSecure: true,
-    },
-    baseUrl: 'http://localhost:8443',
+    detectBrowserLanguage: false,
```
Comment (zackkrida) [inline nuxt.config.js:180]: Now we'd only look at the path supplied to the iframe, no other communication necessary.

**Comment by dhruvkb** [comment]
Wouldn't exposing translations in the URL allow invalid combinations like https://ru.wordpress.org/openverse/es?

**Review by obulat — CHANGES_REQUESTED** [review]
Works great, and will prevent the problems with mismatch with the outer frame we sometimes have (Openverse in French with outer frame in English).
I've scanned the repo for paths, and there is one place where we need to change path to `localePath`:
https://github.com/WordPress/openverse-frontend/blob/0f784f4df9fdf4b460dab387f0877229420efebc/src/components/VAllResultsGrid/VAllResultsGrid.vue#L10

I don't think this change should affect the `urlChange` message exchange with the outer frame in `sendWindowMessage`, but it would be nice to check.

**Comment by obulat** [comment]
> Wouldn't having translations in the URL allow entering invalid combinations like https://ru.wordpress.org/openverse/es?

I guess we could add a redirect to URLs like this. Is it possible to detect the mismatch between the `ru` and `es` locales in the URL `https://ru.wordpress.org/openverse/es`, and then set the iframe locale to `ru`, in this case?

**Comment by zackkrida** [comment]
@dhruvkb it what scenario would a url like ru.wordpress.org/openverse/es be produced? What would be adding the `/es` to the end of the path?

**Comment by dhruvkb** [comment]
It wouldn't be occurring naturally, that's just my imagination of someone trying to mess with the system by intentionally entering such a URL into the searchbar.

**Comment by zackkrida** [comment]
Got it! In this case, I think it would just 404 and that's okay.

**Comment by zackkrida** [comment]
Thanks @obulat I also fixed the 'back to search results' links; I would've missed that without your suggestion!

**Review by obulat — APPROVED** [review]

**Comment by zackkrida** [comment]
Oh also @dhruvkb, only folks using the non-english site who would know to inspect the iframe would even think to look at the url there, notice we're using a base path for locales, and then try to manipulate, where it would only effect their local browsing experience.

File: nuxt.config.js (hunk 1 above)
Comment (sarayourfriend) [reply nuxt.config.js:180]: @zackkrida can you update the comment for this variable? If it's still critical, we should describe why and how, especially now that it's simply disabled.

File: src/composables/use-match-routes.js [hunk 2]
```diff
@@ -1,17 +1,26 @@
...
 export const useMatchRoute = (routes = []) => {
+  const { app } = useContext()
   const route = useRoute()
   const router = useRouter()
-  const matches = ref(routes.includes(route.value.name))
+  const localizedRoutes = routes.map(
+    (route) => app.localeRoute({ name: route }).name
+  )
+  const matches = ref(localizedRoutes.includes(route.value.name))
+  console.log(route.value.name)
```
Comment (sarayourfriend) [inline src/composables/use-match-routes.js:21]:
```suggestion
```

File: src/pages/image/_id.vue [hunk 3]
```diff
@@ -61,7 +61,10 @@ const PhotoDetailPage = {
   },
   beforeRouteEnter(to, from, nextPage) {
     nextPage((_this) => {
-      if (from.path === '/search/' || from.path === '/search/image') {
+      if (
+        from.name === _this.localeRoute({ path: '/search/' }).name ||
+        _this.localeRoute({ path: '/search/image' }).name
+      ) {
```
Comment (sarayourfriend) [inline src/pages/image/_id.vue:67]:
```suggestion
      if (
        from.name === _this.localeRoute({ path: '/search/' }).name ||
        from.name === _this.localeRoute({ path: '/search/image' }).name
      ) {
```

**Review by sarayourfriend — CHANGES_REQUESTED** [review]
Just a couple of things/minor logic bugs.

Otherwise this LGTM. We'll still need one more fix to the homepage to fully fix the internationalization of it as #765 still needs to be addressed.

File: nuxt.config.js (hunk 1 above)
Comment (zackkrida) [reply nuxt.config.js:187]: Thanks for the suggestion, I updated the comment with more useful information!

**Review by sarayourfriend — APPROVED** [review]
LGTM! Thanks for the fixes 🎉

**Pull request merged**